File: calc.c
#include <stdio.h>

float sumar(float x, float y) {
    return x + y;
}

float restar(float x, float y) {
    return x - y;
}

float multiplicar(float x, float y) {
    return x * y;
}

float dividir(float x, float y) {
    if(y != 0.0) {
        return x / y;
    } else {
        printf("Error: División por cero.\n");
        return 0.0;
    }
}

int main() {
    int seleccion;
    float num1, num2, resultado;

    do {
        printf("Operaciones disponibles:\n");
        printf("1. Suma\n");
        printf("2. Resta\n");
        printf("3. Multiplicación\n");
        printf("4. División\n");
        printf("5. Salir\n");
        printf("Selecciona una operación (1/2/3/4/5): ");
        scanf("%d", &seleccion);
        
        if(seleccion == 5) break;

        printf("Ingresa el primer número: ");
        scanf("%f", &num1);
        printf("Ingresa el segundo número: ");
        scanf("%f", &num2);

        switch(seleccion) {
            case 1:
                resultado = sumar(num1, num2);
                printf("Resultado: %.2f\n", resultado);
                break;
            case 2:
                resultado = restar(num1, num2);
                printf("Resultado: %.2f\n", resultado);
                break;
            case 3:
                resultado = multiplicar(num1, num2);
                printf("Resultado: %.2f\n", resultado);
                break;
            case 4:
                resultado = dividir(num1, num2);
                if(num2 != 0.0) {
                    printf("Resultado: %.2f\n", resultado);
                }
                break;
            default:
                printf("Opción inválida. Por favor, intenta de nuevo.\n");
        }
    } while(seleccion != 5);

    printf("Sesion finalizada...\n");
    return 0;
}

File: calc.rs
use std::io;

fn sumar(x: f32, y: f32) -> f32 {
    x + y
}

fn restar(x: f32, y: f32) -> f32 {
    x - y
}

fn multiplicar(x: f32, y: f32) -> f32 {
    x * y
}

fn dividir(x: f32, y: f32) -> Option<f32> {
    if y != 0.0 {
        Some(x / y)
    } else {
        None
    }
}

fn main() {
    loop {
        println!("Operaciones disponibles:");
        println!("1. Suma");
        println!("2. Resta");
        println!("3. Multiplicación");
        println!("4. División");
        println!("5. Salir");
        
        println!("Selecciona una operación (1/2/3/4/5):");
        let mut seleccion = String::new();
        io::stdin().read_line(&mut seleccion).expect("Fallo al leer la línea");
        
        if seleccion.trim() == "5" {
            println!("Sesion finalizada..");
            break;
        }
        
        println!("Ingresa el primer número:");
        let mut num1 = String::new();
        io::stdin().read_line(&mut num1).expect("Fallo al leer la línea");
        let num1: f32 = num1.trim().parse().expect("Por favor, ingresa un número");
        
        println!("Ingresa el segundo número:");
        let mut num2 = String::new();
        io::stdin().read_line(&mut num2).expect("Fallo al leer la línea");
        let num2: f32 = num2.trim().parse().expect("Por favor, ingresa un número");
        
        match seleccion.trim() {
            "1" => println!("Resultado: {}", sumar(num1, num2)),
            "2" => println!("Resultado: {}", restar(num1, num2)),
            "3" => println!("Resultado: {}", multiplicar(num1, num2)),
            "4" => match dividir(num1, num2) {
                Some(resultado) => println!("Resultado: {}", resultado),
                None => println!("Error: División por cero."),
            },
            _ => println!("Opción inválida. Por favor, intenta de nuevo."),
        }
    }
}
